manifests/init.pp

```diff
@@ -476,4 +492,25 @@
   if ($enable_sso) {
     class { 'jira::sso': }
   }
+
+  # install any given library or remove them
+  $plugins.each |Hash $plugin_data| {
+    $target = "${jira::webappdir}/atlassian-jira/WEB-INF/lib/${$plugin_data['installation_name']}"
+    if  $plugin_data['ensure'] == 'absent' {
+      archive {
+        $target:
+          ensure        => 'absent',
+      }
+    } else {
+      archive {
+        $target:
+          ensure        => 'present',
+          source        => $plugin_data['source'],
+          username      => $plugin_data['username'],
+          password      => $plugin_data['password'],
```

Comment:
What happens if the plugin doesn't need username and password and the hash doesn't contain these keys? Maybe we should guard against it and default to `undef`?

Please add a test as well :smile: